### DISCUSSION TITLE
Runs on Android device, Android emulator, and iOS simulator. Crashes immediately on physical iOS device.

### DISCUSSION THREAD

**Discussion by JGKle** [AvaloniaUI/Avalonia#15189] [Q&A] [answered]
I've got a more or less "finished" multi-platform app. It was developed on Android, & works great there. I also have it running in the iOS simulator. I'm developing in Visual Studio on Windows, & paired with a remote Mac, using it as a remote simulator.

I finally just got a physical iOS device, & did all the provisioning/key setup. I'm able to deploy to it & run from Visual Studio, Visual Studio switches to its "Debugging" interface, I see the AvaloniaTest loading screen on the device, and then it immediately throws. I'm running a Debug build. The exception is:

```
ObjCRuntime.ObjCException: 'Objective-C exception thrown.  Name: NSInvalidArgumentException Reason: +[UITraitCollection currentTraitCollection]: unrecognized selector sent to class 0x1d100fb30
```

Any idea why it would fail on the device when it succeeds in the simulator? Note: the only difference is the device is an older version of iOS (12), and Apple will not allow you to download simulators pre-iOS 15. So I can't run a Simulator with the identical version of iOS as the phone, which cannot be upgraded past iOS 12. The app targets iOS 10 though, so it seems like either way, it should work.

Any idea why the simulator would work but the physical device have the exception above?

**Comment by maxkatz6** [comment, marked as answer]
Avalonia supports only 13+ as of right now:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/iOS/Avalonia.iOS/Avalonia.iOS.csproj#L4

It possibly can be extended to 12, but we don't have immediate plans for that.

**Reply by JGKle** [reply]
Dang, well that would explain it. Thanks for the super quick reply.

**Reply by JGKle** [reply]
What about iPadOS, is it the same as iOS? 13+?

**Reply by maxkatz6** [reply]
Yes, same for iPadOS and tvOS.

**Reply by Yveske1966** [reply]
Was going to ask the same question. Thanks for the answer maxkatz6 !
